### ISSUE TITLE
Focal loss with probabilities

### ISSUE THREAD

**Issue by pcicales** [open-mmlab/mmdetection#10849]
**Describe the feature**

**Motivation**
A clear and concise description of the motivation of the feature.

Being able to use focal loss with computed probabilities allows us to ensure stability when we have overflow/under flow that causes probabilities that are near zero for an instance. Currently, only logits are supported through mmengine.

**Related resources**
If there is an official code release or third-party implementations, please also provide the information here, which would be very helpful.

**Additional context**
If you would like to implement the feature and create a PR, please leave a comment here and that would be much appreciated.
